playbook/runner.go

```diff
@@ -168,6 +168,32 @@ func GetActionForAgent(ctx context.Context, agent *models.Agent) (*ActionForAgen
 	return &output, tx.Commit().Error
 }
 
+func checkPlaybookFilter(ctx context.Context, playbook models.Playbook, run models.PlaybookRun) (bool, error) {
+	// Check playbook filter
+	templateEnv, err := prepareTemplateEnv(ctx, run)
+	if err != nil {
+		return false, fmt.Errorf("failed to prepare template env: %w", err)
+	}
+
+	var playbookSpec v1.PlaybookSpec
+	if err := json.Unmarshal(playbook.Spec, &playbookSpec); err != nil {
+		return false, err
+	}
+	for _, f := range playbookSpec.Filters {
+		val, err := ctx.RunTemplate(gomplate.Template{Expression: f}, templateEnv.AsMap())
+		if err != nil {
+			ctx.Errorf("error running playbook[%s] filter expression[%s]: %v", playbook.ID, f, err)
+			return false, nil
+		}
+
+		// The expression must return a boolean
```

Comment:
I think it should support a string return which would be used as the error message

Reply:
Where will we propagate the message ? should we add a details field in playbook_run ?

Reply:
returning evaluated val in err msg now

Reply:
I think we should do explict isTruthy and isFalsey checks and then return "Playbook validation failed" error (Without the playbook id which should be known by the calle), or the string response.